TROUBLESHOOTING.md

```diff
@@ -0,0 +1,155 @@
+# Troubleshooting performance issues
+
+Performance is at the heart of Quarkus.
+
+If you are facing performance issues (runtime or startup issues), and would like to discuss them with the Quarkus Team,
+you are more than welcome on our [mailing list](https://groups.google.com/d/forum/quarkus-dev),
+[Zulip chat](https://quarkusio.zulipchat.com) or [Github issue tracker](https://github.com/quarkusio/quarkus/issues).
+
+To help us to troubleshoot your issues, we will need some performance insight from your application.
+
+On Linux or macOS, one of the best way to gather performance insight would be to generate CPU and allocation [FlameGraphs](https://github.com/brendangregg/FlameGraph) 
+via [Async Profiler](https://github.com/jvm-profiling-tools/async-profiler).
+
+If you want a deeper introduction to Async Profiler, go read [this article](https://hackernoon.com/profiling-java-applications-with-async-profiler-049s2790).
+
+## Installing Async Profiler
+
+To install Async Profiler, go to the [release page](https://github.com/jvm-profiling-tools/async-profiler/releases) and download the latest release.
+
+Async Profiler depends on `perf_events`. <br/>
+To allow capturing kernel call stacks using `perf_events` from a non-root process,
+you must make some OS configuration.
+
+For your session:
+
+```shell script
+echo 1 | sudo tee /proc/sys/kernel/perf_event_paranoid
+echo 0 | sudo tee /proc/sys/kernel/kptr_restrict
+```
+
+Or definitively using `sysctl`:
+
+```shell script
+sudo sysctl -w kernel.perf_event_paranoid=1
+sudo sysctl -w kernel.kptr_restrict=0
+```
+
+For allocation profiling, you also need to install HotStop debug symbol.
+
+Depending on your Linux and Java distribution this can be done via:
```

Comment:
I'm not sure I would recommend that. I think most people don't use the JDK from the distribution.

Reply:
The way to install debug info on a non-distribution JDK is way harder, you need to download it and build it (and building C code can fail in a lot of possible ways).
I can find a link to add but it's way easier to install it this way and use the JDK from the distribution to launch the application.

Reply:
In fact you need to build your own JDK ...
In this case I would suggest to use a fastdebug build, it will be a slower JVM (JVM asserts will be enabled) but it includes debug symbols so it will works.

Reply:
I added a reference to fastdebug builds, didn't include a specific link but I could add one to https://builds.shipilev.net/ it's the only place I know where you can find those builds